lib/msal-browser/docs/testing.md

```diff
@@ -0,0 +1,143 @@
+# Testing
+
+MSAL Browser starting version 2.17.0 has added the `loadTokens()` API, which allows the loading of id tokens and access tokens to the MSAL cache, which can then be read out using `acquireTokenSilent()`. 
+
+**Note: This is an advanced feature that is intended for testing purposes only. Loading tokens to your application's cache may cause your app to break.**
+
+## The loadTokens() API
+
+The `loadToken()` API can be accessed by calling `getTokenCache()` on MSAL Browser's `PublicClientApplication` instance. 
+
+```js
+myMSALObj.getTokenCache().loadTokens(silentRequest, serverResponse, loadTokenOptions);
+```
+
+`loadTokens()` takes in a request of type `SilentRequest`, a response of type `ServerAuthenticationROPCRespones`, and options of type `LoadTokenOptions`.
+
+See the type definitions for each below:
+
+- [`SilentRequest`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_browser.html#silentrequest)
+- [`ServerAuthenticationROPCResponse`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_common.html#serverauthorizationropcresponse)
+
+```js
+export type LoadTokenOptions = {
+    clientInfo?: string,
+    extendedExpiresOn?: number,
+    callback?: TokenCallback
+};
+```
+
+### Loading id tokens
+
+Provide the following to load an id token:
+
+1. A server response with the id token, scopes, token_type, and expires_in value, and
+1. Either:
+    - A `SilentRequest` object with account information, OR
+    - A `SilentRequest` object with the authority AND a `LoadTokenOptions` object with `clientInfo`, OR
+    - A `SilentRequest` object with the authority AND a server response object with `client_info`
+
+See the code examples below:
+
+```js
+const silentRequest: SilentRequest = {
+    scopes: ["User.Read", "email"],
+    account: {
+        homeAccountId: "your-home-account-id",
+        environment: "login.microsoftonline.com",
+        tenantId: "your-tenant-id",
+        username: "test@contoso.com",
+        localAccountId: "your-local-account-id"
+    }
+};
+
+const serverResponse: ServerAuthorizationROPCResponse = {
+    token_type: "Bearer",
+    scopes: "User.Read email",
+    expires_in: 3599,
+    id_token: "id-token-here"
+}
+
+const loadTokenOptions: LoadTokenOptions = {};
+
+OR
+
+const silentRequest: SilentRequest = {
+    scopes: ["User.Read", "email"],
+    authority: "https://login.microsoftonline.com/your-tenant-id"
+};
+
+const serverResponse: ServerAuthorizationROPCResponse = {
+    token_type: "Bearer",
+    scopes: "User.Read email",
+    expires_in: 3599,
+    id_token: "id-token-here",
+}
+
+const loadTokenOptions: LoadTokenOptions = {
+    clientInfo: "client-info-here"
+};
+
+OR
+
+const silentRequest: SilentRequest = {
+    scopes: ["User.Read", "email"],
+    authority: "https://login.microsoftonline.com/your-tenant-id"
+};
+
+const serverResponse: ServerAuthorizationROPCResponse = {
```

Comment:
Curious: Is there a use case where the server response contains only `idToken`? My understanding is that ROPC fetches all three. Of course it is also scope dependent?

Reply:
Or is this we giving options to the users to be able to pass any combo of these tokens?

Reply:
As far as I know, ROPC fetches all three, but we are scoping `loadTokens` to just id tokens and access tokens at the moment, with access tokens being optional. Users can pass in just an id token, or an id token and an access token.

Reply:
Got it. Can we please call this out in the docs? Basically when users make their service call, they will get the `refreshToken`. We want them to understand we are intentionally not accepting `refreshToken` for side loading tokens.

Reply:
Good call-out. Done.